plugin/src/main/java/org/opensearch/ml/action/controller/CreateControllerTransportAction.java

```diff
@@ -100,37 +100,41 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLCrea
             ActionListener<MLCreateControllerResponse> wrappedListener = ActionListener.runBefore(actionListener, context::restore);
             mlModelManager.getModel(modelId, null, excludes, ActionListener.wrap(mlModel -> {
                 FunctionName functionName = mlModel.getAlgorithm();
+                Boolean isHidden = mlModel.getIsHidden();
                 if (functionName == TEXT_EMBEDDING || functionName == REMOTE) {
                     modelAccessControlHelper
                         .validateModelGroupAccess(user, mlModel.getModelGroupId(), client, ActionListener.wrap(hasPermission -> {
                             if (hasPermission) {
                                 if (mlModel.getModelState() != MLModelState.DEPLOYING) {
                                     indexAndCreateController(mlModel, controller, wrappedListener);
+                                } else {
+                                    final String errorMessage =
+                                        "Creating a model controller during its corresponding model in DEPLOYING state is not allowed, please either create the model controller after it is deployed or before deploying it.";
```

Comment:
@b4sjoo  Can you help review ? Seems this error message is enough. Do we really need to return the model id in error message ?

Reply:
This PR adds a lot of if/else checking for just handling error message of hidden model. Looks hard to maintain in future. How about we just use a common error message for both hidden and normal models

Reply:
If you are referring to this errorMsg, then cx should know already for which model he/she is creating the controller.

But in general for more debugging purposes, I kept model ID as much as possible for the normal model.